bundles/framework/layerselector2/service/layerlist.js

```diff
@@ -23,6 +24,32 @@
             getName: function () {
                 return this.__name;
             },
+            getMutator () {
+                const me = this;
+                return {
+                    setCurrentFilter (filterId) {
+                        me.currentFilter = me.currentFilter === filterId ? null : filterId;
+                        if (me.currentFilter) {
+                            // Set current filter style to active and others to deactive
+                            const currentFilter = Object.values(me.layerlistFilterButtons).filter(b => b.id === me.currentFilter)[0];
+                            currentFilter.cls.current = currentFilter.cls.active;
+                            const otherFilters = Object.values(me.layerlistFilterButtons).filter(b => b.id !== me.currentFilter);
+                            if (otherFilters) {
+                                otherFilters.map(o => {
+                                    o.cls.current = o.cls.deactive;
+                                    return o;
+                                });
+                            }
+                        } else {
+                            Object.values(me.layerlistFilterButtons).map(o => {
+                                o.cls.current = o.cls.deactive;
+                                return o;
+                            });
+                        }
```

Comment:
Create a new array for layerlistFilterButtons to enable shallow comparisation.

Reply:
Changed as suggested.